apps/vc-api/docs/tutorials/consent-and-issuance-credential-tutorial.md

```diff
@@ -0,0 +1,1701 @@
+
+**Request URL**
+
+`{VC API base url}/vc-api/exchanges`
+
+**HTTP Verb**
+
+`POST`
+
+**Request Body**
+
+Fill `exchangeId` in the json below.
+`exchangeId` should be an id unique to this exchange, for example a UUID.
+
+Note the constraint on the `$.type` path of the credential.
+This is used to require that the presented credentials are of type "PermanentResidentCard" and "ConsentCredential".
+For further documentation regarding the `presentationDefinition`, can be seen [here](../exchanges.md#presentation-definition-queries)
+
+In order to test the notification functionality, you can use the "[webhook.site](https://webhook.site/)".
+This is a free website which allows you to view and debug callback/webhook HTTP POST requests.
+With this service, requests are saved to a dedicated location for later review.
+Please only use this service for this tutorial (or other non-production applications).
+
+To use the webhook.site service with this tutorial, use a dedicated endpoint url generated for you after entering
+the site. It should looke similar to `https://webhook.site/efb19fb8-2579-4e1b-8614-d5a03edaaa7a`
+Copy this URL, including the domain, into the exchange definition below.
+
+```json
+{
+   "exchangeId":"<SOME UNIQUE ID>",
+   "query":[
+         {
+            "type":"PresentationDefinition",
+            "credentialQuery":[
+               {
+                  "presentationDefinition":{
+                     "id":"286bc1e0-f1bd-488a-a873-8d71be3c690e",
+                     "input_descriptors":[
+                        {
+                           "id":"PermanentResidentCard",
+                           "name":"PermanentResidentCard",
+                           "purpose":"PermanentResidentCard",
+                           "constraints":{
+                              "fields":[
+                                 {
+                                    "path":[
+                                       "$.type"
+                                    ],
+                                    "filter":{
+                                       "type":"array",
+                                       "contains":{
+                                          "type":"string",
+                                          "const":"PermanentResidentCard"
+                                       }
+                                    }
+                                 }
+                              ]
+                           }
+                        }
+                     ]
+                  }
+               }
+            ]
+         },
+         {
+            "type":"PresentationDefinition",
+            "credentialQuery":[
+               {
+                  "presentationDefinition":{
+                     "id":"286bc1e0-f1bd-488a-a873-8d71be3c690e",
+                     "input_descriptors":[
+                        {
+                           "id":"consent_credential",
+                           "name":"Consent Credential",
```

Comment:
The consent credential presentation definition should use the `subject_is_issuer` constraint, IMO. https://identity.foundation/presentation-exchange/#relational-constraint-feature